internal/loggers/pid_logger.go

```diff
@@ -0,0 +1,49 @@
+package loggers
+
+import (
+	"fmt"
+	"github.com/DIMO-Network/edge-network/internal/api"
+	"github.com/DIMO-Network/edge-network/internal/queue"
+	"github.com/google/uuid"
+	log "github.com/sirupsen/logrus"
+	"sync"
+)
+
+//go:generate mockgen -source pid_logger.go -destination mocks/pid_logger_mock.go
+type PIDLogger interface {
+	ExecutePID(header, mode, pid, formula, protocol string) (err error)
+}
+
+type pidLogger struct {
+	mu           sync.Mutex
+	unitID       uuid.UUID
+	storageQueue queue.StorageQueue
+}
+
+func NewPIDLogger(unitID uuid.UUID, storageQueue queue.StorageQueue) PIDLogger {
+	return &pidLogger{unitID: unitID, storageQueue: storageQueue}
+}
+
+func (vl *pidLogger) ExecutePID(header, mode, pid, formula, protocol string) (err error) {
```

Comment:
thoughts on just passing in entire pidConfig object? That way you'll also have the signal name